Search/Factory.php

```diff
@@ -49,13 +53,37 @@ public function makeField($name, $value, $type = Field::TYPE_STRING)
     }
 
     /**
-     * Make a new metadata object representing a mapped
-     * searchable class
+     * Make a new metadata object repesenting a mapped
+     * searchanle class
+     */
+    public function makeClassMetadata($class)
+    {
+        return new ClassMetadata($class);
+    }
+
+    /**
+     * Make a new metadata object representing an index for
+     * a mapped searchable class.
      *
      * @return IndexMetadata
      */
-    public function makeIndexMetadata($class)
+    public function makeIndexMetadata()
+    {
+        return new IndexMetadata();
+    }
+
+    public function makeMetadataField($name)
+    {
+        return new MetadataField($name);
+    }
+
+    public function makeMetadataProperty($path)
```

Comment:
Ok, that's fine for me. But I think it is uncommon to have one factory class for all of them... But the classes are connected in some way, so it seems to be ok. However, don't you think that `create` is a more common name than `make`?